Become plugin tests fail on devel

### Summary

https://dev.azure.com/ansible/b24bf3ca-6168-45d7-99e2-bf8029e67c87/_apis/build/builds/16166/logs/1631
https://dev.azure.com/ansible/b24bf3ca-6168-45d7-99e2-bf8029e67c87/_apis/build/builds/16167/logs/1586
```

2021-05-26T07:45:11.0771432Z 07:25 =================================== FAILURES ===================================
2021-05-26T07:45:11.0772027Z 07:25 [31m[1m_________________________________ test_sudosu __________________________________[0m
2021-05-26T07:45:11.0772556Z 07:25 [gw1] linux -- Python 3.8.0 /usr/bin/python3.8
2021-05-26T07:45:11.0773144Z 07:25 mocker = <pytest_mock.plugin.MockerFixture object at 0x7f53f934f1f0>
2021-05-26T07:45:11.0773903Z 07:25 parser = ArgumentParser(prog='testparser', usage=None, description=None, formatter_class=<class 'ansible.cli.arguments.option_helpers.SortingHelpFormatter'>, conflict_handler='resolve', add_help=True)
2021-05-26T07:45:11.0774373Z 07:25 reset_cli_args = None
2021-05-26T07:45:11.0774597Z 07:25 
2021-05-26T07:45:11.0774869Z 07:25     def test_sudosu(mocker, parser, reset_cli_args):
2021-05-26T07:45:11.0775176Z 07:25         options = parser.parse_args([])
2021-05-26T07:45:11.0775471Z 07:25         context._init_global_context(options)
2021-05-26T07:45:11.0775757Z 07:25         play_context = PlayContext()
2021-05-26T07:45:11.0775977Z 07:25     
2021-05-26T07:45:11.0776210Z 07:25         default_cmd = "/bin/foo"
2021-05-26T07:45:11.0776473Z 07:25         default_exe = "/bin/bash"
2021-05-26T07:45:11.0776853Z 07:25         sudo_exe = 'sudo'
2021-05-26T07:45:11.0777253Z 07:25         sudo_flags = '-H -s -n'
2021-05-26T07:45:11.0777475Z 07:25     
2021-05-26T07:45:11.0777768Z 07:25 >       cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
2021-05-26T07:45:11.0778329Z 07:25 E       AttributeError: 'PlayContext' object has no attribute 'make_become_cmd'
2021-05-26T07:45:11.0778625Z 07:25 
2021-05-26T07:45:11.0778878Z 07:25 tests/unit/plugins/become/test_sudosu.py:27: AttributeError
```

### Issue Type

Bug Report

### Component Name

become plugins unit tests

### Ansible Version

devel

### Configuration

*

### OS / Environment

*

### Steps to Reproduce

*

### Expected Results

*

### Actual Results

*

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

--- a/tests/unit/plugins/become/test_sudosu.py
+++ b/tests/unit/plugins/become/test_sudosu.py
@@ -10,36 +10,41 @@ __metaclass__ = type
 import re
 
 from ansible import context
-from ansible.playbook.play_context import PlayContext
-from ansible.plugins.loader import become_loader
+
+from .helper import call_become_plugin
 
 
 def test_sudosu(mocker, parser, reset_cli_args):
     options = parser.parse_args([])
     context._init_global_context(options)
-    play_context = PlayContext()
 
     default_cmd = "/bin/foo"
     default_exe = "/bin/bash"
     sudo_exe = 'sudo'
     sudo_flags = '-H -s -n'
 
-    cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
-    assert cmd == default_cmd
-
     success = 'BECOME-SUCCESS-.+?'
 
-    play_context.become = True
-    play_context.become_user = 'foo'
-    play_context.set_become_plugin(become_loader.get('community.general.sudosu'))
-    play_context.become_flags = sudo_flags
-    cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
-
-    assert (re.match("""%s %s  su -l %s %s -c 'echo %s; %s'""" % (sudo_exe, sudo_flags, play_context.become_user,
+    task = {
+        'become_user': 'foo',
+        'become_method': 'community.general.sudosu',
+        'become_flags': sudo_flags,
+    }
+    var_options = {}
+    cmd = call_become_plugin(task, var_options, cmd=default_cmd, executable=default_exe)
+    print(cmd)
+    assert (re.match("""%s %s  su -l %s %s -c 'echo %s; %s'""" % (sudo_exe, sudo_flags, task['become_user'],
                                                                   default_exe, success, default_cmd), cmd) is not None)
 
-    play_context.become_pass = 'testpass'
-    cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
+    task = {
+        'become_user': 'foo',
+        'become_method': 'community.general.sudosu',
+        'become_flags': sudo_flags,
+        'become_pass': 'testpass',
+    }
+    var_options = {}
+    cmd = call_become_plugin(task, var_options, cmd=default_cmd, executable=default_exe)
+    print(cmd)
     assert (re.match("""%s %s -p "%s" su -l %s %s -c 'echo %s; %s'""" % (sudo_exe, sudo_flags.replace('-n', ''),
-                                                                         r"\[sudo via ansible, key=.+?\] password:", play_context.become_user,
+                                                                         r"\[sudo via ansible, key=.+?\] password:", task['become_user'],
                                                                          default_exe, success, default_cmd), cmd) is not None)

